ogr/ogrsf_frmts/generic/ogr_gensql.cpp

```diff
@@ -1339,7 +1345,56 @@ OGRFeature *OGRGenSQLResultsLayer::TranslateFeature(OGRFeature *poSrcFeat)
     swq_evaluation_context sContext;
     for (int iField = 0; iField < psSelectInfo->result_columns(); iField++)
     {
-        swq_col_def *psColDef = &psSelectInfo->column_defs[iField];
+        const swq_col_def *psColDef = &psSelectInfo->column_defs[iField];
+
+        if (psColDef->bHidden)
+        {
+            if ((psColDef->field_alias &&
+                 EQUAL(psColDef->field_alias, "OGR_STYLE")) ||
+                (!psColDef->field_alias &&
+                 EQUAL(psColDef->field_name, "OGR_STYLE")))
+            {
+                if (psColDef->field_type == SWQ_STRING)
+                {
+                    if (psColDef->field_index != -1)
```

Comment:
changed to:
```
                    // Does this column definition directly references a
                    // source field ?
                    if (psColDef->field_index >= 0)
```

Reply:
That helps for this particular case. Maybe outside the scope of this PR but it might be useful to add comments to property names back in the definition of the `swq_col_def` struct. Could also consider adding trivial helper methods like `bool referencesSourceField() const { return field_index >= 0; }`.